### PR TITLE
Do not crash on ASCII85 in inline images and properly support their colorspaces

### LINKED ISSUE #1008: inline image parsing fails when stream data contains "EI\n"
In the case where an inline image uses the ASCII85Decode filter, it can (and frequently does) contain the sequence "EI" internally at the end of a line.  This confuses the pdf parser and can lead to a variety of weird symptoms, since it will attempt to parse the rest of the stream data as the rest of the containing content stream, which it most definitely is not.  I just happened to notice this because the attached PDF has this problem (the symptom in particular is that it comes across the sequence `"\611"` which is a very invalid octal escape):

[222-2008-zonage-annexe-c-carte-25b-innond.pdf](https://github.com/user-attachments/files/16167527/222-2008-zonage-annexe-c-carte-25b-innond.pdf)

The PDF spec is not tremendously helpful, but once you realize that an ASCII85Decode stream *must* end with `"~>"` it's obvious that we should be looking for that, and not "EI" followed by whitespace, in the case where this encoding is used.  This should be as simple as checking if `/A85` is in the image dictionary and then passing `"~>"` instead of "EI" to `PDFContentParser.get_inline_data`.  I'll make a PR.

### PR DESCRIPTION
**Pull request by dhdaines**
**Pull request**

ASCII85Decode was not properly handled in inline images (because the end tag "EI\n" is very frequently present in ASCII85 streams) causing random crashing.

Also inline images weren't properly being extracted as BMP because we weren't recognizing their abbreviated colorspaces.

Fixes #1008

**How Has This Been Tested?**

There's a unit test ;-)

**Checklist**

- [ x ] I have read [CONTRIBUTING.md](https://github.com/pdfminer/pdfminer.six/blob/master/CONTRIBUTING.md). 
- [ X ] I have added a concise human-readable description of the change to [CHANGELOG.md](https://github.com/pdfminer/pdfminer.six/blob/master/CHANGELOG.md).
- [ x ] I have tested that this fix is effective or that this feature works.
- [ x ] I have added docstrings to newly created methods and classes.
- [ x ] I have updated the [README.md](https://github.com/pdfminer/pdfminer.six/blob/master/README.md) and the [readthedocs](https://github.com/pdfminer/pdfminer.six/tree/master/docs/source) documentation. Or verified that this is not necessary.

### CODE REVIEW

File: pdfminer/pdfinterp.py [hunk 1]
```diff
@@ -331,11 +332,22 @@ def do_keyword(self, pos: int, token: PSKeyword) -> None:
...
-                (pos, data) = self.get_inline_data(pos + len(b"ID "))
+                d = {literal_name(k): resolve1(v) for (k, v) in choplist(2, objs)}
+                eos = b"EI"
+                filter = d.get("F", None)
+                if filter is not None:
+                    if isinstance(filter, PSLiteral):
+                        filter = [filter]
+                    for f in LITERALS_ASCII85_DECODE:
+                        if f in filter:
+                            eos = b"~>"
```
Comment (pietermarsman) [inline pdfminer/pdfinterp.py:343]: I'm wondering if this is correct if ASCII85 is not the first filter. In other words, has a FlateDecode ASCII85 stream the same end-of-stream marker as an ASCII85 FlateDecode stream. I suspect the first filter determines the end-of-stream marker.

**Review by pietermarsman — CHANGES_REQUESTED** [review]
Hi, 

Thanks for this contribution! It is looking good but I need some more time to understand the details of this change.

File: pdfminer/pdfinterp.py (hunk 1 above)
Comment (dhdaines) [reply pdfminer/pdfinterp.py:343]: Ah, good catch.  Exactly, it should only apply if ASCII85 is the first filter (in the case of this specific document it is followed by FlateDecode).  I'll update in a second!

**Review by pietermarsman — APPROVED** [review]
I've updated the comments and the CHANGELOG a bit. 

Thanks for fixing this!

**Pull request merged**